### PR TITLE
Remove npc highlights on task completed message.

### LINKED ISSUE #5414: Slayer target highlighting not clearing on task completion

### PR DESCRIPTION
**Pull request by MackBryan**
Closes #5414

Simple fix, they wanted highlights to clear after they have killed the final monster on task. Previously it would only remove them once a new task was received from a slayer master.

Not sure if entirely useful or wanted but only took a minute so do with it as you please.

### CODE REVIEW

**Comment by MackBryan** [comment]
@ShaunDreclin Yeah I didn't look into it much but that looks to be it. Let me know if you want me to change anything.

**Comment by ShaunDreclin** [comment]
Move the null check from `setTask()` to inside `rebuildTargetNames()` below the `targetNames.clear()` line, then you should be able to remove the extra `highlightedTargets.clear()` from the section that detects task completion.

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 1]
```diff
@@ -520,12 +520,16 @@ private boolean isTarget(NPC npc)
 	private void rebuildTargetNames(Task task)
 	{
 		targetNames.clear();
-		Arrays.stream(task.getTargetNames())
-			.map(String::toLowerCase)
-			.forEach(targetNames::add);
-		targetNames.add(taskName.toLowerCase().replaceAll("s$", ""));
-	}
+		highlightedTargets.clear();
```
Comment (ShaunDreclin) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: This line is unnecessary, there's already a `highlightedTargets.clear()` in `rebuildTargetList()` which gets called right after this function

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 2]
```diff
@@ -520,12 +520,16 @@ private boolean isTarget(NPC npc)
...
 
+		if (task != null)
+		{
+			Arrays.stream(task.getTargetNames())
+					.map(String::toLowerCase)
+					.forEach(targetNames::add);
+			targetNames.add(taskName.toLowerCase().replaceAll("s$", ""));
+		}
+	}
```
Comment (ShaunDreclin) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: Need a blank line between these two functions

**Comment by ShaunDreclin** [comment]
uhh I think that commit you just pushed undid the changes you made haha

**Comment by MackBryan** [comment]
Yup it did haha, sorry its late and I am exhausted. New one should have everything fixed up. Anyways I have an interview in the morning so if there is anything else I missed I will get it after that.

**Pull request merged**